examples/tutorial_oss.py

```diff
@@ -4,45 +4,25 @@
 import torch
 import torch.distributed as dist
 import torch.multiprocessing as mp
-import torch.nn as nn
-import torch.nn.functional as F
 
 from fairscale.optim.oss import OSS
+from helpers import dist_init, getModel, getData, getLossFun
 
 WORLD_SIZE = 2
 EPOCHS = 3
 
 DEVICE = "cuda" if torch.cuda.is_available() else "cpu"
 
 
-def dist_init(rank, world_size):
-    backend = dist.Backend.NCCL if torch.cuda.is_available() else dist.Backend.GLOO  # type: ignore
-    print(f"Using backend: {backend}")
-    dist.init_process_group(backend=backend, init_method="tcp://localhost:29501", rank=rank, world_size=world_size)
-
-
-def getModel():
-    return nn.Sequential(torch.nn.Linear(10, 10), torch.nn.ReLU(), torch.nn.Linear(10, 5))
-
-
-def getData():
-    target = torch.randint(0, 2, size=(20, 1)).squeeze()
-    data = torch.randn(20, 10)
-    return [(data, target)]
-
-
-def getLossFun():
-    return F.nll_loss
-
-
 def train(rank: int, world_size: int, epochs: int, use_oss: bool):
 
     # DDP
     dist_init(rank, world_size)
+    rank = torch.device("cpu") if DEVICE == "cpu" else rank
 
     # Problem statement
     model = getModel().to(rank)
-    dataloader = getData()
+    dataloader = [getData()]
```

Comment:
hmm the name was not super explicit here, more `getDataloader`, but it's a bit more typical to get an iterator here than just a [list], no ? Else the loop down below will not go very far..